torch/nn/_intrinsic/qat/modules/conv_fused.py

```diff
@@ -0,0 +1,265 @@
+            bound = 1 / math.sqrt(fan_in)
+            init.uniform_(self.bias, -bound, bound)
+        # A hack to avoid resetting on undefined parameters
+        if hasattr(self, 'gamma'):
+            self.reset_bn_parameters()
+
+    def enable_fake_quant(self):
+        self.observer.enable()
+        self.weight_fake_quant.enable()
+        return self
+
+    def disable_fake_quant(self):
+        self.observer.disable()
+        self.weight_fake_quant.disable()
+        return self
+
+    def _forward(self, input):
+        # exponential_average_factor is self.momentum set to
+        # (when it is available) only so that if gets updated
+        # in ONNX graph when this node is exported to ONNX.
+        if self.momentum is None:
+            exponential_average_factor = 0.0
+        else:
+            exponential_average_factor = self.momentum
+
+        if self.training and self.track_running_stats:
+            # TODO: if statement only here to tell the jit to skip emitting this when it is None
+            if self.num_batches_tracked is not None:
+                self.num_batches_tracked += 1
+                if self.momentum is None:  # use cumulative moving average
+                    exponential_average_factor = 1.0 / float(self.num_batches_tracked)
+                else:  # use exponential moving average
+                    exponential_average_factor = self.momentum
+        scale_factor = self.gamma / torch.sqrt(self.running_var + self.eps)
+        scaled_weight = self.weight * scale_factor.reshape([-1, 1, 1, 1])
+        conv = self.conv2d_forward(input, self.weight_fake_quant(scaled_weight))
+
+        print('bias:', self.bias)
+        if self.bias is not None:
+            conv_orig = (conv - self.bias.reshape([1, -1, 1, 1])) / scale_factor.reshape([1, -1, 1, 1])
+        else:
+            conv_orig = conv / scale_factor.reshape([1, -1, 1, 1])
+        print('conv_orig:', conv_orig)
+        batch_mean = torch.mean(conv_orig, dim=[0, 2, 3])
+        if self.bias is not None:
+            batch_mean = batch_mean + self.bias
+        batch_var = torch.var(conv_orig, dim=[0, 2, 3], unbiased=False)
+
+        if not self.freeze_bn:
+            rescale_factor = torch.sqrt((self.running_var + self.eps) / (batch_var + self.eps))
+            batch_rstd = 1.0 / torch.sqrt(batch_var + self.eps)
+            conv = conv * rescale_factor.reshape([1, -1, 1, 1])
+            conv = conv + (self.beta - self.gamma * batch_mean * batch_rstd).reshape([1, -1, 1, 1])
+            if self.bias is not None:
+                conv = conv + ((self.gamma * batch_rstd - rescale_factor) * self.bias).reshape([1, -1, 1, 1])
+        else:
+            conv = conv + (self.beta - self.gamma * self.running_mean /
+                           torch.sqrt(self.running_var + self.eps)).reshape([1, -1, 1, 1])
+
+        self.running_mean = exponential_average_factor * batch_mean + (1 - exponential_average_factor) * self.running_mean
```

Comment:
running mean and var should be updated only if training is True and freeze_bn is False:
```
if self.training and not self.freeze_bn:
      self.running_mean = exponential_average_factor * batch_mean + (1 - exponential_average_factor) * self.running_mean       
      self.running_var = exponential_average_factor * batch_var + (1 - exponential_average_factor) * self.running_var
```

Reply:
yeah, just did that